### ISSUE TITLE
Is it possible to expand a capture group into a custom type when using NativeTestCase?

### ISSUE THREAD

**Issue by cameroncooke** [net-a-porter-mobile/XCTest-Gherkin#146]
This is simplistic example but proves my point.

Given I have a feature file like:

```gherkin
Scenario Outline: My test outline
    Given I enter the name <name>

    Examples:
    | name       |
    | Jill Smith |
```

and the step definition:

```swift
step("I enter the name (.+)") { (match: Person) in 
    ...
}
```

and the model:

struct Person: CodableMatchedStringRepresentable {
    let name: String
    ...
    func doSomethingToPerson() {
        ...
    }
}

In the above example the expression will be `I enter the name (Jill Smith)` and I would like to turn the String "Jill Smith" from the capture group into the type `Person` in my step defintion.

I note that you can do this when using native Swift based feature steps as shown in your README by using resolving the instance into JSON via the description property.

```swift
Given("User is loggeed in as \(Person(name: "Nick"))")
```

I assume what happens here is `Person(name: "Nick")` is encoded into JSON so the expression becomes: `"User is loggeed in as ({"name": "Nick"}))"` and then when the test is ran it encodes that back into the `Person` type?

In assume in my situation I would have to write the feature file like:

```gherkin
Scenario Outline: My test outline
    Given I enter the name {"name": "<name>"}

    Examples:
    | name       |
    | Jill Smith |
```

But this is less than ideal. 

Is there another way I achive this? Thanks.

**Comment by ilyapuchka** [comment]
@cameroncooke yes, in swift tests json string is encoded as part of the step expression and then decoded from this string. It might work if you write json structure in a feature file if you implement step so that it uses `Codable` parameter, but this was not tested specifically (we can add tests for that if it really works). 

I think more proper solution though would be to support data tables - I had it working for swift tests, don't remember if I did anything for feature files, but I'll need to find time to get back to this.
